metrics/ci_integrations/lib/client/github_actions/github_actions_client.dart

```diff
@@ -0,0 +1,319 @@
+    int runId, {
+    int perPage = 10,
+    int page,
+  }) {
+    final queryParameters = {
+      'per_page': '$perPage',
+      'page': '$page',
+    };
+
+    final url = UrlUtils.buildUrl(
+      basePath,
+      path: 'runs/$runId/artifacts',
+      queryParameters: queryParameters,
+    );
+
+    return _fetchRunArtifacts(url);
+  }
+
+  /// Retrieves the next [WorkflowRunArtifactsPage] of the given [currentPage].
+  Future<InteractionResult<WorkflowRunArtifactsPage>> fetchNextRunArtifactsPage(
+    WorkflowRunArtifactsPage currentPage,
+  ) {
+    return _fetchRunArtifacts(currentPage.nextPageUrl);
+  }
+
+  /// Retrieves [WorkflowRunArtifactsPage] by the given [url].
+  ///
+  /// Both [fetchRunArtifacts] and [fetchNextRunArtifactsPage] delegate
+  /// retrieving [WorkflowRunArtifactsPage]s to this method.
+  Future<InteractionResult<WorkflowRunArtifactsPage>> _fetchRunArtifacts(
+    String url, {
+    int page,
+    int perPage,
+  }) {
+    return _handleResponse<WorkflowRunArtifactsPage>(
+      _client.get(url, headers: headers),
+      (Map<String, dynamic> json, Map<String, String> headers) {
+        final nextPageUrl = parseNextPageUrl(headers);
+
+        final totalCount = json == null ? null : json['total_count'] as int;
+
+        final artifactsList =
+            json == null ? null : json['artifacts'] as List<dynamic>;
+        final artifacts = WorkflowRunArtifact.listFromJson(artifactsList);
+
+        return InteractionResult.success(
+          result: WorkflowRunArtifactsPage(
+            totalCount: totalCount,
+            page: page,
+            perPage: perPage,
+            nextPageUrl: nextPageUrl,
+            values: artifacts,
+          ),
+        );
+      },
+    );
+  }
+
+  /// Fetches a workflow run artifact by the given download [url].
+  ///
+  /// The resulting [Uint8List] contains bytes of a zip archive with the desired
+  /// artifacts.
+  Future<InteractionResult<Uint8List>> downloadRunArtifactZip(
+    String url,
+  ) async {
+    try {
+      final response = await _client.get(url);
+
+      if (response.statusCode == HttpStatus.ok) {
+        return InteractionResult.success(result: response.bodyBytes);
+      } else {
+        final reason = response.body == null || response.body.isEmpty
+            ? response.reasonPhrase
+            : response.body;
+        return InteractionResult.error(
+          message: 'Failed to perform an operation with code '
+              '${response.statusCode}. Reason: $reason',
+        );
+      }
+    } catch (error) {
+      return InteractionResult.error(
+        message: 'Failed to perform an operation. Error details: $error',
+      );
+    }
+  }
+
+  /// Parses the next page URL from the given [headers].
+  String parseNextPageUrl(Map<String, String> headers) {
+    final linkHeader = headers['link'];
+    final nextLinkString = linkHeader.split(',').firstWhere(
+          (link) => link.contains('next'),
```

Comment:
```suggestion
          (link) => link.contains('rel="next"'),
```
Should we be more strict here as the other parts of the header can contain the `next` pattern as well?